<?xml version="1.0"?>
<launch>
    <arg name="fcu_url" default="udp://127.0.0.1:14551@:14551" />
    <arg name="gcs_url" default="" />
    <arg name="tgt_system" default="$(env PX4_SYS_ID)" />
    <arg name="tgt_component" default="1" />
    <arg name="log_output" default="screen" />
    <arg name="fcu_protocol" default="v2.0" />
    <arg name="respawn_mavros" default="false" />

    <include file="$(find mavros)/launch/node.launch">
        <arg name="pluginlists_yaml" value="$(find mavros)/launch/px4_pluginlists.yaml" />
        <arg name="config_yaml" value="$(find mavros)/launch/px4_config.yaml" />
        <arg name="fcu_url" value="$(arg fcu_url)" />
        <arg name="gcs_url" value="$(arg gcs_url)" />
        <arg name="tgt_system" value="$(arg tgt_system)" />
        <arg name="tgt_component" value="$(arg tgt_component)" />
        <arg name="log_output" value="$(arg log_output)" />
        <arg name="fcu_protocol" value="$(arg fcu_protocol)" />
        <arg name="respawn_mavros" default="$(arg respawn_mavros)" />
    </include>

    <!-- Flight Patterns
	0: CIRCULAR
	1: SPIRAL
	2: CLOUD (frequency 7.0 for clear pattern)
	3: SINE (speed 0.3, frequency 3.0 for displacement of 1.26)
	4: N-GRAM (hepta-gram, vertices 7)
	-->
	<arg name="flight_pattern" default="2"/>
	<arg name="max_iter" default="2"/>
	<!-- Flight Parameters -->
	<arg name="dt" default="0.05"/>
	<arg name="radius" default="0.80"/>
	<arg name="height" default="1.00"/>
	<arg name="speed" default="0.30"/>
	<arg name="min_speed" default="0.05"/>
	<arg name="offset_x" default="0.00"/>
	<arg name="offset_y" default="0.00"/>
	<arg name="offset_z" default="0.50"/>
	<arg name="frequency" default="7.00"/>
	<arg name="ngram_vertices" default="7"/>
	<arg name="ngram_step" default="2"/>

    <node pkg="offboard_test" exec="offboard_test_node" name="offboard_test_node" output="screen">
    	<param name="flight_pattern" value="$(var flight_pattern)" />
		<param name="max_iter" value="$(var max_iter)" />
		<param name="height" value="$(var height)" />
		<param name="dt" value="$(var dt)" />
		<param name="radius" value="$(var radius)" />
		<param name="speed" value="$(var speed)" />
		<param name="min_speed" value="$(var min_speed)" />
		<param name="offset_x" value="$(var offset_x)" />
		<param name="offset_y" value="$(var offset_y)" />
		<param name="offset_z" value="$(var offset_z)" />
		<param name="frequency" value="$(var frequency)" />
		<param name="ngram_vertices" value="$(var ngram_vertices)" />
		<param name="ngram_step" value="$(var ngram_step)" />
	</node>    
</launch>
